### ISSUE TITLE
cloud-aws different keys for EC2 & S3

### ISSUE THREAD

**Issue by nowakan** [elastic/elasticsearch#19445]
**Elasticsearch version**:  2.3.2

**JVM version**: openjdk version "1.8.0_91"
                          OpenJDK Runtime Environment (build 1.8.0_91-b14)
                          OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)

**OS version**: Red Hat Enterprise Linux Server release 7.2 (Maipo)

**Description of the problem including expected versus actual behavior**:
Can't provide two access & secret keys (different for ec2 and s3).
I'd like to use cloud-aws plugin for EC2 Discovery & S3 Repository (compatible service).
I've got separate cloud accounts for Eucalyptus & RiakS3.

**Steps to reproduce**:
1) Add to elasticsearch.yml:
```
cloud:
    aws:
        s3:
            access_key: xxxxxxxxxxxx
            secret_key: XXxxxxxXxxxxXxxxXxXxXxxx
            endpoint: s3_cloud_front.address.here
            protocol: http
            signer: S3SignerType
        ec2:
            access_key: yyyyyyyyyyyyy
            secret_key: yyyYyyyyyYyyyyYyyyYyYyY
            endpoint: ec2_cloud_front.address.here
            protocol: http

discovery:
    type: ec2
    ec2:
        groups: group_name
```
2) # /etc/init.d/elasticsearch restart

**Provide logs (if relevant)**:
`[2016-07-14 15:28:17,690][INFO ][discovery.ec2 ] [Hideko Takata] Exception while retrieving instance list from AWS API: Unable to load AWS credentials from any provider in the chain`

**Comment by dadoonet** [comment]
It's a bug to me. I need to reproduce it.

**Comment by dadoonet** [comment]
I read the code today and that's indeed a bug in 2.x series. 

We never read values for [EC2](https://github.com/elastic/elasticsearch/blob/2.4/plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/AwsEc2Service.java#L27-L27) but we do for [S3](https://github.com/elastic/elasticsearch/blob/2.4/plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java#L65-L68).

I'm pretty sure it's ok for 5.0 (I still need to check it though).
I'm starting working on a fix for 2.x series.

As a workaround, I think you could use something like:

```yml
cloud:
    aws:
        access_key: yyyyyyyyyyyyy
        secret_key: yyyYyyyyyYyyyyYyyyYyYyY
        s3:
            access_key: xxxxxxxxxxxx
            secret_key: XXxxxxxXxxxxXxxxXxXxXxxx
            endpoint: s3_cloud_front.address.here
            protocol: http
            signer: S3SignerType
        ec2:
            endpoint: ec2_cloud_front.address.here
            protocol: http

discovery:
    type: ec2
    ec2:
        groups: group_name
```

ec2 will use the "global" AWS access key but S3 will use the S3 specific one.

**Comment by dadoonet** [comment]
Closed by #19513 in 2.4 branch.
